compiler/include/baseAST.h

```diff
@@ -546,12 +550,23 @@ static inline const CallExpr* toConstCallExpr(const BaseAST* a)
     AST_CALL_CHILD(_a, ForwardingStmt, toFnDef, call, __VA_ARGS__);     \
     break;                                                              \
   case E_TryStmt:                                                       \
-    AST_CALL_CHILD(_a, TryStmt, _body, call, __VA_ARGS__);             \
+    AST_CALL_CHILD(_a, TryStmt, _body, call, __VA_ARGS__);              \
     AST_CALL_LIST(_a, TryStmt, _catches, call, __VA_ARGS__);            \
     break;                                                              \
   case E_CatchStmt:                                                     \
-    AST_CALL_CHILD(_a, CatchStmt, _body, call, __VA_ARGS__);           \
+    AST_CALL_CHILD(_a, CatchStmt, _body, call, __VA_ARGS__);            \
     break;                                                              \
+  case E_ForallStmt: {                                                  \
+    AST_CALL_LIST (_a, ForallStmt,    iterExprs(),  call, __VA_ARGS__); \
+    ForallIntents* fi = ((ForallStmt*)_a)->withClause();                \
+    AST_CALL_STDVEC(fi->fiVars,  Expr,              call, __VA_ARGS__); \
+    AST_CALL_STDVEC(fi->riSpecs, Expr,              call, __VA_ARGS__); \
+    AST_CALL_CHILD(fi, ForallIntents, iterRec,      call, __VA_ARGS__); \
+    AST_CALL_CHILD(fi, ForallIntents, leadIdx,      call, __VA_ARGS__); \
+    AST_CALL_CHILD(fi, ForallIntents, leadIdxCopy,  call, __VA_ARGS__); \
+    AST_CALL_CHILD(_a, ForallStmt,    masterBlock(),call, __VA_ARGS__); \
```

Comment:
In other words, how about simply calling `AST_CALL_CHILD` on the `fWith` field (via withClause() accessor, given that the field is private)? Instead of saving it in `fi` and going over each of fi's fields individually?

The answer is that we cannot/should not define AST_CALL_CHILD on the fWith field -- because it contains a ForallIntent object, which is not in the BaseAST hierarchy. I can define a AST_CALL_WITH_CLAUSE macro instead and use it here, if that is preferable.

One of my TODOs is to bring ForallIntent into the BaseAST hierarchy, which will make fWith be an AList of those things. That will resolve this issue.